pkg/v13/kubeconfig/desired.go

```diff
@@ -0,0 +1,60 @@
+package kubeconfig
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/kubeconfig"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/client/k8srestconfig"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/cluster-operator/pkg/label"
+	"github.com/giantswarm/cluster-operator/pkg/v13/chartconfig"
+)
+
+func (r *StateGetter) GetDesiredState(ctx context.Context, clusterConfig chartconfig.ClusterConfig) ([]*corev1.Secret, error) {
```

Comment:
@rossf7 Here I get `ClusterConfig` already, do you think it's okay?

Reply:
No I don't think this will work. There are a few problems including chartconfig.ClusterConfig is the wrong package.

Like I suggested in our call I would create the resource  in `service/controller/aws/v14/resource/kubeconfig`. Get it working there first with the generic secret resource.

The shared code can then be moved to `pkg/v14/kubeconfig` and you can add the azure and kvm resources.